pyronear/models/ssresnet.py

```diff
@@ -0,0 +1,115 @@
+from torchvision.models.resnet import ResNet, resnet18, BasicBlock
+import torchvision
+from torch import nn
+import torch
+import torch.nn.functional as F
+
+
+__all__ = ["SSResNet"]
+
+
+class SSResNet(ResNet):
+    """This model is designed to be trained using the SubSamplerDataSet. It can be built over any resnet.
+    The SubSamplerDataSet will send within the same batch K consecutive frames belonging to the same
+    sequence. The SSresnet model will process these K frames independently in the first 4 layers of
+    the resnet then combine them in a 5th layer.
+    Attributes
+    ----------
+    frame_per_seq: int
+        Number of frame per sequence
+    To build a Resnet we need two arguments, are we using a BasicBlock or a Bottleneck and
+    the corresponding layers. This is how to build the ResNets:
+    resnet18: BasicBlock, [2, 2, 2, 2]
+    resnet34: BasicBlock, [3, 4, 6, 3]
+    resnet50: Bottleneck, [3, 4, 6, 3]
+    resnet101: Bottleneck, [3, 4, 23, 3]
+    resnet152: Bottleneck, [3, 8, 36, 3]
+    Please refere to torchvision documentation for more details:
+    https://github.com/pytorch/vision/blob/master/torchvision/models/resnet.py#L232
+    block: string
+        block: BasicBlock or Bottleneck
+    layers: list
+        layers argument to build BasicBlock / Bottleneck
+    pretrainedWeights: bool
+        pretrained Weights for the ResNet
+    Then we need shapes of the layer5
+    intput1L5: int
+        Input of the layer5, must be equal to the output of layer4
+    output1L5: int
+        Output shape of the first conv1x1
+    output2L5: int
+        Output shape of the second conv1x1
+    """
+    def __init__(self, block, layers, frame_per_seq=2, intput1L5=512,
+                 output1L5=512, output2L5=256):
```

Comment:
Not exactly sure, but the `512` is actually here because the last conv layer of resnet18 has 512 channels. If so, I'd suggest to retrieve dynamically that value using this https://github.com/pytorch/vision/blob/master/torchvision/models/resnet.py#L158

It's actually `512 * block.expansion`, so you don't need `input1L5` or `output1L5`. Regarding `output2L5`, perhaps renaming it into something more user-friendly would be beneficial ;)

Reply:
Ok tanhks for that, I have fixed it :)